src/aggregator/aggregator/entry.go

```diff
@@ -696,13 +696,17 @@ func (e *Entry) updateStagedMetadatasWithLock(
 func (e *Entry) addUntimedWithLock(timestamp time.Time, mu unaggregated.MetricUnion) error {
 	var err error
 	for i := range e.aggregations {
+		ts := timestamp
 		resendEnabled := e.aggregations[i].resendEnabled
 		// Migrate an originally untimed metric (server timestamp) to a "timed" metric (client timestamp) if
 		// resendEnabled is set on the rollup rule. Continuing to use untimed allows for a seamless transition since
 		// the Entry does not change.
-		if resendEnabled && mu.ClientTimeNanos > 0 {
+		if mu.ClientTimeNanos == 0 {
+			resendEnabled = false
+		}
+		if resendEnabled {
```

Comment:
figured i'd also fix another theoretical problem where `resendEnabled` is set on the pipeline but the client timestamp isn't provided.